### PR TITLE
Jest 27

### PR DESCRIPTION
**Pull request by NathanPB**
I updated this to work with the current latest major version of jest =) Seems to be working fine

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,8 +4,8 @@
   "description": "Run benchmark with Jest",
   "main": "dist/index.js",
   "scripts": {
-    "test": "npm run build && jest --projects jest-node.config.ts jest-jsdom.config.ts jest-electron.config.ts",
-    "test:node": "npm run build && jest --projects jest-node.config.ts",
```
Comment (pckhoi) [inline package.json]: Please change this back, this is needed to test the reporter.

File: test/error.bench.ts [hunk 2]
```diff
@@ -1,7 +1,8 @@
-import { benchmarkSuite } from "../dist";
+import { benchmarkSuite } from "../src";
+import {expect} from "@jest/globals";
 
 benchmarkSuite("error", {
   ["should throw error"]: () => {
-    throw new Error("something wrong!");
+    expect(() => { throw new Error("something wrong!") }).toThrow();
```
Comment (pckhoi) [inline test/error.bench.ts]: This need to throw so that I can see the test report the error.

File: test/basic.bench.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import { benchmarkSuite } from "../dist";
+import { benchmarkSuite } from "../src";
```
Comment (pckhoi) [inline test/basic.bench.ts]: Please change this back. Import from dist more accurately represent usage of this library.

File: test/setup_teardown.bench.ts [hunk 4]
```diff
@@ -1,4 +1,4 @@
-import { benchmarkSuite } from "../dist";
+import { benchmarkSuite } from "../src";
```
Comment (pckhoi) [inline test/setup_teardown.bench.ts]: This.

**Review by pckhoi — CHANGES_REQUESTED** [review]
Hello, thank you for the pull request. However, I see this error when I try to test. Are you seeing this?
```

 FAIL  test/setup_teardown.bench.ts
  ● Test suite failed to run

      ● Test suite failed to run

        TypeError: Cannot read property 'bind' of undefined

      at jestAdapter (node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapter.js:30:56)
```
It seems there is something breaking alright. Can you please take care of this?

**Comment by NathanPB** [comment]
Ima check that all in a few days, for my experience it looks like jest-circus is kinda broken, couldn't find a solution yet

**Comment by NathanPB** [comment]
The requested changes are done, just had to revert one commit. Jest-circus still broken tho

**Comment by pckhoi** [comment]
Looks like the issue is that jest-electron still stucks at jest@v24: https://github.com/hustcc/jest-electron/issues/32

Let's see how that goes.

**Comment by NathanPB** [comment]
Ye, but this issue is more than a year old, I don't think it's gonna happen anytime soon... I think we will be stuck with jest 24 forever

For the time being I published my fork on npmjs because I needed this library for my personal use. It's on ``@nathanpb/jest-bench`` (if anyone reading this in the future need it)

**Comment by pckhoi** [comment]
I have publish `jest-bench@27.3.1` for those who want to use jest version 27. From now on jest-bench will take its version after jest's. Anyone who is still stuck at jest@26 can install `jest-bench@26.6.3`.

**Pull request closed without merging**